lesson01/assignment/scripts.js

```diff
@@ -30,4 +30,129 @@ document. addEventListener("DOMContentLoaded", function(){
+    const woodchucks = ['Chucky', 'Chuckanut', 'Chuckalog', 'Chuckerella', 'Phil'];
+
+    // Present the array as an unordered list. Can I turn this into an arrow function?
+    woodchucks.forEach(chuckers);
+
+    function chuckers(item) {
+      document.querySelector('#woodchuck-list').innerHTML += '<li>' + item + '</li>';
+    }
+
+    // User picks a name from the list
+    const chuckForm = document.querySelector('#chuck-form');
+
+    chuckForm.addEventListener('submit', function(event) {
+
+      event.preventDefault();
+
+    // Turning user input into variables
+    const input3 = document.querySelector('#that-guy');
+    const badchuck = input3.value;
+
+    document.querySelector('#bad-chuck').innerHTML = badchuck;
+
+    input3.value = '';
+
+  });
+
+    // This form will get user input to make an array from a string
+    const pigLatin = document.querySelector('#pig-latin');
+
+    pigLatin.addEventListener('submit', function(event) {
+
+      event.preventDefault();
+
+      const input4 = document.querySelector('#word-to-mangle');
+      const wordBefore = input4.value.toUpperCase();
+
+      // Create the array, and remove the first item
+      let arrayWord = Array.from(wordBefore);
+      let firstLetter = arrayWord.shift();
+
+      // Concatenate arrays
+      let step4 = arrayWord.concat(firstLetter);
+
+      // Turn them back into a string
+      let step5 = step4.join('');
+
+      document.querySelector('#plt').innerHTML = step5 + 'AY';
+
+      input4.value = '';
+
+  });
+// This last one doesn't work properly but I decided that I need to just hand it in and move on.
+  // Get surprise smoothie ingredients by picking the number of letters in the ingedient name
+
+  const recipe = document.querySelector('#recipe');
+
+  recipe.addEventListener('submit', function(event) {
+
+    event.preventDefault();
+
+    // Wow, gross
+    let ingredients = [
+      {name: 'mango', length: 5},
+      {name: 'chips', length: 5},
+      {name: 'vodka', length: 5},
+      {name: 'banana', length: 6},
+      {name: 'potato', length: 6},
+      {name: 'salmon', length: 6},
+      {name: 'apricot', length: 7},
+      {name: 'pumpkin', length: 7},
+      {name: 'clothes', length: 7},
+      {name: 'cucumber', length: 8},
+      {name: 'cherries', length: 8},
+      {name: 'nitrogen', length: 8},
+      {name: 'raspberry', length: 9},
+      {name: 'asparagus', length: 9},
+      {name: 'detergent', length: 9},
+      {name: 'watermelon', length: 10},
+      {name: 'strawberry', length: 10},
+      {name: 'maybeetles', length: 10},
+    ];
+
+    // Convert user input to integer for filtering list
+    const input5 = document.querySelector('#name-length');
+    const nameLength = input5.value;
+    const nameLengthInt = parseInt(nameLength);
+
+    // find items with length matching user input
+    const smoothie = ingredients.filter( ({length}) => length === nameLengthInt);
+    console.log(smoothie);
```

Comment:
You are really close here. The reason you're not seeing the results come out right is you can not assign objects to HTML. If `smoothie` or `ingredients` were arrays of strings, you could absolutely set HTML to that and you'd see the text correctly.

What you'd need to do here is find a way to extract the name of each object in your `const smoothie` variable.

Lucky for us, there is a really nice javascript function that lets us do this. What you'd want to do is something like this:

```   
const smoothieIngredients = smoothie.map(ingredientObject => ingredientObject.name)

// Display results
document.querySelector('#ingredient-list').innerHTML = smoothieIngredients;
```

What this map function does is loop through `smoothie`, which is all the objects you just filtered. Then, it creates a new array with the value you choose to extract. In this case, for each ingredient in `smoothie` we want to extract the name.

So we end up with an array of strings that we can then assign to HTML!